woltka/workflow.py

```diff
@@ -0,0 +1,590 @@
+                raise ValueError(errmsg)
+            files = {path_: sample}
+            samples = [sample]
+
+    # path is a directory of alignment files
+    elif isdir(path_):
+
+        # turn off demultiplexing if not decided
+        demux = demux or False
+
+        # get a map of plausible sample Ids to files
+        map_ = id2file_map(path_, ext, not demux and samples)
+        if len(map_) == 0:
+            raise ValueError('No valid file found in directory.')
+        if demux:
+            files = sorted([join(path_, x) for x in map_.values()])
+
+        # validate with given sample Ids
+        else:
+            if not samples:
+                samples = sorted(map_.keys())
+            elif len(map_) < len(samples):
+                raise ValueError(errmsg)
+            files = {join(path_, map_[x]): x for x in samples}
+
+    else:
+        raise ValueError(f'"{path_}" is not a valid file or directory.')
+
+    click.echo(f'Alignment files to read: {len(files)}.')
+    click.echo(f'Demultiplexing: {"on" if demux else "off"}.')
+    return samples, files, demux
+
+
+def build_align_proc(coords_fp: str = None,
+                     overlap:   int = None) -> object:
+    """Build alignment processor.
+
+    Parameters
+    ----------
+    coords_fp : str, optional
+        Path to gene coordinates file.
+    overlap : int, optional
+        Read/gene overlapping percentage threshold
+
+    Returns
+    -------
+    object
+        Alignment processor.
+
+    Notes
+    -----
+    Currently two processors are supported: Plain() for regular alignments
+    (i.e., simple query-to-subject maps), Ordinal() for alignments with
+    coordinates which will be used to match queries (reads) and genes. The
+    presence of a gene coordinates file (`coords_fp`) is an indicator for
+    using the latter.
+    """
+    if coords_fp:
+        click.echo(f'Reading gene coordinates...', nl=False)
+        with readzip(coords_fp) as fh:
+            coords = read_gene_coords(fh, sort=True)
+        click.echo(' Done.')
+        click.echo(f'Total host sequences: {len(coords)}.')
+        return Ordinal(coords, whether_prefix(coords),
+                       overlap and overlap / 100)
+    else:
+        return Plain()
+
+
+def build_hierarchy(names_fp:   str = None,
+                    nodes_fp:   str = None,
+                    newick_fp:  str = None,
+                    lineage_fp: str = None,
+                    ranktb_fp:  str = None,
+                    map_fps:   list = None,
+                    map_rank:  bool = False) -> (dict, dict, dict, str):
+    """Construct hierarchical classification system.
+
+    Parameters
+    ----------
+    names_fp : str, optional
+        Taxonomic names file.
+    nodes_fp : str, optional
+        Taxonomic nodes file.
+    newick_fp : str, optional
+        Newick tree file.
+    lineage_fp : str, optional
+        Lineage strings file.
+    ranktb_fp : str, optional
```

Comment:
```suggestion
    rank_table_fp : str, optional
```

Reply:
And corresponding refactorings.

Reply:
For this particular question I don't have a very good answer. There should be a better way to describe "a table with each row as a taxon and each column as a rank and columns are ordered by hierarchy from high to low". The phrase "rank table" isn't most meaningful. Do you have an alternative suggestion? @gwarmstrong

Reply:
This is the first time I recall having seen a requirement for "ordered by hierarchy from high to low" in the documentation... I do not think this requirement is in the readme?

You have a good question, as there is certainly a trade-off.... I guess this brings to mind, it seems like this format will be really spares? Why do you not take some other format that is essentially just a map? e.g., `taxa\trank`?

Reply:
Currently README says: "Table of per-taxon per-rank assignments. Each column represents a rank. Column header will be treated as rank name." You are right that no spec of high-to-low is in place. I plan to elaborate README on this.

I am also thinking of removing this feature, since it is confusing and, less common and, essentially similar to Greengenes-style lineage strings and can be achieved by multiple maps. What do you think?

Reply:
Can you elaborate on the "multiple maps" solution? I do not quite understand what you mean.